[Kaptain] requirements section (#4471)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.0.0/install/requirements/index.md

---
layout: layout.pug
navigationTitle: Requirements
title: Requirements
menuWeight: 3
excerpt: Requirement reference values
beta: false
enterprise: false
---

The amount of resources required for your project **depends on the workload** you want to run, you need to consider: 

-   The number, complexity, and size of your training jobs, notebook servers, deployed models, etc.
-   The amount of metadata and log data stored with each run.

> To estimate the amount of resources required for your project, it is recommended that you test your specific use case.

Author: @bruced2iq, @dmennell is there anything else here we could mention to help customers? How do customers currently estimate their resources? Maybe we could include 1 or 2 lines on that here?

To help you in this task, the following table provides an account of the resources required for two common operations: [installing Kaptain][install], and running a complete machine learning workflow using Tensorflow. This workflow corresponds to the [Kaptain SDK with Tensorflow tutorial][tensorflow], and includes the training, tuning, and deploying stages of a machine learning model used to recognize handwritten digits.

> The resources in the table are required exclusively for each specified task, and assume you have previously allocated other resources to install and run DKP. Running any real world machine learning workload on Kaptain bumps these requirements. For this reason, the number of CPUs, GPUs, RAM, and persistent disk space **must be increased considerably**. 

Reviewer: ```suggestion
The resources in the table are required exclusively for each specified task, and assume you have previously allocated other resources to install and run DKP.   These are minimums; 'real-world' machine learning workloads will require substantially more of all the resource types.
```

Consider the following use cases as reference values:

| Cluster resource       | Kaptain installation | Kaptain SDK with Tensorflow tutorial |
|------------------------|----------------------|--------------------------------------|
| Number of worker nodes | at least 3           | at least 3                           |
| CPU per node           | 8 cores              | 16 cores                             |

> | RAM per node           | 16 GiB               | 64 GiB                               |

Reviewer 2: ```suggestion
| RAM per node           | 16 GiB               | 32 GiB                               |
```

Author: And this applies to 2.0.0 and 2.1.0, correct?

Reviewer 3: Yes

> | Storage per node       | 60 GiB               | 60 GiB                               |

Author: @alembiewski @andresmascl as suggested by Dan, is it ok if we bump this to:
```suggestion
| Storage per node       | 120 GiB               | 120 GiB                               |
```

> 
> | Cluster resource       | Kaptain installation | Kaptain SDK with Tensorflow tutorial |
> |------------------------|----------------------|--------------------------------------|
> | Number of worker nodes | at least 3           | at least 3                           |
> | CPU per node           | 8 cores              | 24 cores                             |
> | RAM per node           | 16 GiB               | 86 GiB                               |
> | Storage per node       | 60 GiB               | 61 GiB                               |
> 

Reviewer 3: based on @andresmascl's [measurements](https://docs.google.com/spreadsheets/d/1VoF4MCsiUD2NnogcPZ-2wSsco1auZY-j91o_nHZPxMY/edit#gid=1219806816), I can see CPU requests are doubled and memory requests are x4.  Storage increase by 1 GB is rather confusing and can be ignored. Adjusting the numbers to the closest "power of two" values:
```suggestion

| Cluster resource       | Kaptain installation | Kaptain SDK with Tensorflow tutorial |
|------------------------|----------------------|--------------------------------------|
| Number of worker nodes | at least 3           | at least 3                           |
| CPU per node           | 8 cores              | 16 cores                             |
| RAM per node           | 16 GiB               | 64 GiB                               |
| Storage per node       | 60 GiB               | 60 GiB                               |

```

<p class="message--note"><strong>NOTE: </strong>Only NVIDIA GPU instances are supported.</p>

> The [Kaptain installation][install] example illustrates the resources required to install Kaptain and its dependencies without running any type of workload. In other words, the **bare minimum** resources. These resources must be **freely available** within the cluster for the installation to succeed.

Reviewer: ```suggestion
The [Kaptain installation][install] column lists the resources required to install Kaptain and its dependencies without running any workload. In other words, the **bare minimum** amount of resources. These resources must be **freely available** within the cluster for the installation to succeed.
```

> The [Kaptain SDK with Tensorflow tutorial][tensorflow] example illustrates the resources required to install Kaptain and run the Kaptain SDK with Tensorflow tutorial. Again, these resources must be **freely available** within the cluster, and should not be allocated to any other applications.

Reviewer: ```suggestion
The [Kaptain SDK with Tensorflow tutorial][tensorflow] column the resources required to install Kaptain and run the Kaptain SDK with Tensorflow tutorial. Again, these resources must be **freely available** within the cluster, and should not be allocated to any other applications.
```

> For **on premises** installations, horizontal scalability is limited by the overall size of the cluster and its quotas.

Reviewer 4: ```suggestion
For **on-premises** installations, horizontal scalability is limited by the overall size of the cluster and its quotas.
```

> For **on premise** installations, horizontal scalability is limited by the overall size of the cluster and its quotas.

Reviewer: ```suggestion
For **on premises** installations, horizontal scalability is limited by the overall size of the cluster and its quotas.
```
